mantle/cmd/kola/qemuexec.go

```diff
@@ -41,21 +49,100 @@ var (
 	ignition string
 	knetargs string
 
+	ignitionFragments []string
+
 	forceConfigInjection bool
 )
 
 func init() {
 	root.AddCommand(cmdQemuExec)
 	cmdQemuExec.Flags().StringVarP(&knetargs, "knetargs", "", "", "Arguments for Ignition networking on kernel commandline")
 	cmdQemuExec.Flags().BoolVarP(&usernet, "usernet", "U", false, "Enable usermode networking")
+	cmdQemuExec.Flags().StringSliceVar(&ignitionFragments, "add-ignition", nil, "Append well-known Ignition: [\"autologin\"]")
 	cmdQemuExec.Flags().StringVarP(&hostname, "hostname", "", "", "Set hostname via DHCP")
 	cmdQemuExec.Flags().IntVarP(&memory, "memory", "m", 0, "Memory in MB")
 	cmdQemuExec.Flags().StringVarP(&ignition, "ignition", "i", "", "Path to ignition config")
 	cmdQemuExec.Flags().BoolVarP(&forceConfigInjection, "inject-ignition", "", false, "Force injecting Ignition config using guestfs")
 }
 
+func renderFragments() (string, error) {
+	buf, err := ioutil.ReadFile(ignition)
+	if err != nil {
+		return "", err
+	}
+	config, _, err := v3.Parse(buf)
+	for _, fragtype := range ignitionFragments {
+		switch fragtype {
+		case "autologin":
+			newconf := v3.Merge(config, conf.GetAutologin())
+			config = newconf
+			break
+		default:
+			return "", fmt.Errorf("Unknown fragment: %s", fragtype)
+		}
+	}
+
+	newbuf, err := json.Marshal(config)
+	if err != nil {
+		return "", err
+	}
+	tmpf, err := ioutil.TempFile("", "qemuexec-ign")
+	if err != nil {
+		return "", err
+	}
+	if _, err := tmpf.Write(newbuf); err != nil {
+		return "", err
+	}
+
+	return tmpf.Name(), nil
+}
+
 func runQemuExec(cmd *cobra.Command, args []string) error {
 	var err error
+	cleanupIgnition := false
+	if len(ignitionFragments) > 0 {
+		newconfig, err := renderFragments()
+		if err != nil {
+			return errors.Wrapf(err, "rendering fragments")
+		}
+		ignition = newconfig
+		cleanupIgnition = true
+	}
+	if kola.Options.IgnitionVersion == "v2" {
```

Comment:
Minor: just so I understand, this if-statement isn't related to the autologin draining, right? Feels like it's worth its own commit.

Reply:
It's *necessary* now because previously `cosa run` worked with RHCOS, but `kola qemuexec` required external translation.

Reply:
We have a similar version of this down translation code in a few places. Might end up being worthwhile to move more into the `Conf` / `UserData` struct and providing a stable interface to perform translation from there.